services/blockstorage/sync/factory.go

```diff
@@ -2,24 +2,79 @@ package sync
 
 import (
 	"github.com/orbs-network/orbs-network-go/instrumentation/log"
+	"github.com/orbs-network/orbs-network-go/instrumentation/metric"
 	"github.com/orbs-network/orbs-spec/types/go/primitives"
 	"github.com/orbs-network/orbs-spec/types/go/protocol/gossipmessages"
 	"github.com/orbs-network/orbs-spec/types/go/services/gossiptopics"
+	"time"
 )
 
 type stateFactory struct {
 	config  blockSyncConfig
 	gossip  gossiptopics.BlockSync
 	storage BlockSyncStorage
+	c       *blockSyncConduit
 	logger  log.BasicLogger
+	metrics *stateMetrics
 }
 
-func NewStateFactory(config blockSyncConfig, gossip gossiptopics.BlockSync, storage BlockSyncStorage, logger log.BasicLogger) *stateFactory {
+type stateMetrics struct {
+	carStateLatency           *metric.Histogram
```

Comment:
what's the difference between this and the next metric?

Reply:
its a work in progress, apologies for not mentioning it, right now the collecting (car) finished and  waiting are only measting the state transition latency, need to add more metrics there but did not want to invest more efforts in case i am doing something wrong